### PR TITLE
[Rust] Implement project discovery and Cargo metadata output

### LINKED ISSUE #52: [Rust] Implement project discovery and Cargo metadata output
## Goal
Implement project-mode discovery for Cargo manifests and stable package metadata output.

## Scope
- Scan the input directory for `Cargo.toml`
- Fill output metadata fields:
  - `cargoTomlPath`
  - `numOfCargoToml`
  - `packageInfo`
  - `moduleName`
- Add synthetic project fixtures and golden tests for discovery output

## Out of scope
- Rust syntax AST-to-UAST lowering
- Full workspace semantics beyond required metadata fields

## Acceptance criteria
- Project mode finds Cargo manifests under `-rootDir`
- Metadata fields are present and stable across repeated runs
- Golden tests for discovery output pass
- `cargo test` passes

## Dependencies
- Parent planning issue: #50
- Depends on: #51

### PR DESCRIPTION
**Pull request by KinomotoMio**
## Summary
Implement project-mode Cargo manifest discovery and stable metadata output for the Rust parser.

## Scope in this PR
- Discover `Cargo.toml` files recursively in project mode
- Fill project metadata fields from discovery results:
  - `moduleName`
  - `cargoTomlPath`
  - `numOfCargoToml`
  - `packageInfo` tree
- Keep output deterministic via stable manifest sorting
- Add multi-manifest synthetic fixtures and golden regression tests
- Add edge-case fixes:
  - skip hidden directories during discovery
  - parse quoted package names safely when `#` appears inside quotes

## Out of scope
- Rust AST -> UAST syntax lowering
- Engine/analyzer/taint integration

## Validation
- `cargo fmt --all`
- `cargo test` (pass)
- Golden and stability checks for project mode output (pass)

## Related issues
- Part of #50
- Closes #52

## Note
This is stacked on top of #56.

### CODE REVIEW

File: parser-Rust/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+fn manifest_rel_path(root_dir: &Path, manifest_path: &Path) -> String {
+    let rel = manifest_path
+        .strip_prefix(root_dir)
+        .unwrap_or(manifest_path);
+    let rel_str = rel.to_string_lossy().replace('\\', "/");
+    if rel_str.starts_with('/') {
+        rel_str
+    } else {
+        format!("/{rel_str}")
+    }
```
[the remark this answers is not in the document]
Comment (KinomotoMio) [reply parser-Rust/src/lib.rs]: Handled: `manifest_rel_path` now uses `strip_prefix(...).expect("discovered manifest should be inside root_dir")` and simplified path formatting.

**Comment by KinomotoMio** [comment]
Closing in favor of #56.

Rust parser progress is now consolidated on branch `feat/rust-parser` and tracked in #56.
Planning/tracking is also consolidated in #50.

Note: this is currently exploratory work and does not guarantee mergeability yet.
